### ISSUE TITLE
Private space creation fails with M_LIMIT_EXCEEDED error

### ISSUE THREAD

**Issue by BillCarsonFr** [vector-im/element-web#23620]
### Steps to reproduce

1. Where are you starting? What can you see?
Simple Space creation fails
<img width="521" alt="image" src="https://user-images.githubusercontent.com/9841565/198084279-6ad11b8f-6654-4351-af01-2dcbf202e689.png">

3. What do you click?

Just create a new account, and started the flow to create a private space.

#### What did you expect?

That it worked
Android is failing too. 
So probably some server change regarding room creation?

### Homeserver

matrix.org

### Will you send logs?

Yes

**Comment by turt2live** [comment]
This appears to be picked up by the backend team already - have assigned a maintainer from there to take a look.

**Comment by BillCarsonFr** [comment]
Ok, to be complete this failure has an annoying side effect if you retry:

<img width="323" alt="image" src="https://user-images.githubusercontent.com/9841565/198244720-04e51910-8386-4a9c-bf34-aa37db300326.png">


^ you end up with a bunch of rooms with no names and not attached to any space.

Space creation is not atomic

**Comment by reivilibre** [comment]
Even on older Synapse versions, such as 1.68.0 (before the most recent regression where the rate limiting is much harsher — too harsh), if you fill in all 3 boxes quickly right after creating the space, you are faced with an error and all the 3 rooms are created but one of them isn't made a `m.space.child` of the space (though it has the `m.space.parent` event in the child room, so it shows in the space whilst active — just not on the 'Explore space' screen — mighty confusing but oh well.).
I think the client would benefit from some sensible retry logic when rate limiting occurs.

I also notice that if the initial space creation is rate limited, the dialogue stays open and does not close:
<img width="428" alt="image" src="https://user-images.githubusercontent.com/38398653/198363126-28aec011-91a9-43d3-b828-c7a696cb5bbd.png">


TL;DR: it's true that Synapse is being a bit harsh, especially v1.69.0+ where the harshness is several times what is normal, but the client could do with better handling for rate limit exceeded errors.

**Comment by reivilibre** [comment]
The issue in Synapse is now closed and we have made the rate limiter more generous to allow for this use case, but note that it's still possible for a user to exhaust their ratelimiter tokens by sending a lot of messages beforehand, right before creating a space. The client should have appropriate safeguards and retry logic for this.
